### ISSUE TITLE
Support the full FormControl API

### ISSUE THREAD

**Issue by gparlakov** [gparlakov/forms-typed#38]
Support the AbstractFormControlOptions:

```ts
     validatorOrOpts?: ValidatorFn | ValidatorFn[] | AbstractControlOptions | null, 
     asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
```

**Comment by gparlakov** [comment]
Add the types of the emissions 
```ts
 * Type encapsulating the Angular Form options:
 * `emitEvent` - do we emit event;
 * `onlySelf` - do we bubble up to parent
 * `emitModelToViewChange` - When true or not supplied (the default), each change triggers an onChange event to update the view.
 * `emitViewToModelChange` - When true or not supplied (the default), each change triggers an ngModelChange event to update the model.
 */
export interface FormEventOptions {
    emitEvent?: boolean;
    onlySelf?: boolean;
    emitModelToViewChange?: boolean;
    emitViewToModelChange?: boolean;
}
```

**Comment by Mongole666** [comment]
Handing ValidatorFn[] and AsyncValidatorFn[] seems to be done by just adding these to the constructor call in the types. Or am I missing something here? So can this issue be closed with a new minor version in the near future? @gparlakov

**Comment by gparlakov** [comment]
@Mongole666 yes - you are correct, I just saw that the `typedFormControl` does not support that while `typedFormGroup` and `typedFormArray` do... Will definitely do a minor release with that, thanks.

In the mean time `Validators.compose` and `Validators.composeAsync` can do the trick...

`const form = typedFormControl<number>(1, Validators.compose([Validators.required, Validators.minLength(1)]), Validators.composeAsync([....]))`

**Comment by gparlakov** [comment]
@Mongole666 just released version 1.1.2 with support for that
